pennylane/tape/jacobian_tape.py

```diff
@@ -685,6 +685,36 @@ def hessian(self, device, params=None, **options):
             # parameters. Simply return an empty Hessian.
             return np.zeros((len(params), len(params)), dtype=float)
 
+        # The parameter-shift Hessian implementation currently only supports
+        # the two-term parameter-shift rule. Raise an error for unsupported operations.
+        supported_ops = (
+            "RX",
+            "RY",
+            "RZ",
+            "Rot",
+            "PhaseShift",
+            "ControlledPhaseShift",
+            "MultiRZ",
+            "PauliRot",
+            "U1",
+            "U2",
+            "U3",
+            "SingleExcitationMinus",
+            "SingleExcitationPlus",
+            "DoubleExcitationMinus",
+            "DoubleExcitationPlus",
+        )
+
+        for idx, info in self._par_info.items():
+            op = info["op"]
+
+            if idx in self.trainable_params and op.name not in supported_ops:
+                raise ValueError(
+                    f"The operation {op.name} is currently not supported for the "
+                    f"parameter-shift Hessian.\nPlease decompose the operation in your "
+                    f"QNode by replacing it with '{op.__str__().replace('(', '.decomposition(')}'"
```

Comment:
Is it a problem that we are already transitioning to `expand` instead of `decomposition` in operations?

Reply:
This error message should only be for this release, since we are moving where the internal expansion happens for v0.16 🙂 

So even if we don't manage to generalize the Hessian parameter-shift by v0.16, at the very least the Hessian parameter-shift rule will be able to perform this decomposition internally (see #1264)